packages/core-components/src/components/toggle-switch/toggle-switch.tsx

```diff
@@ -0,0 +1,97 @@
+import {
+  Component,
+  Host,
+  Element,
+  Prop,
+  h,
+  Event,
+  EventEmitter,
+} from '@stencil/core';
+import { ToggleSwitchEventDetail } from '../../utils/interfaces/interaction.interface';
+
+@Component({
+  tag: 'b2b-toggle-switch',
+  styleUrl: 'toggle-switch.scss',
+  shadow: true,
+})
+export class B2bToggleSwitchComponent {
+  @Element() host: HTMLB2bToggleSwitchElement;
+
+  /** The label of the toggle button. This is required */
+  @Prop() label!: string;
+
+  /** The alignment of the toggle switch label. */
+  @Prop() labelPosition?: 'left' | 'right' = 'left';
+
+  /** Whether or not the toggle button is currently disabled. Per default it is false. */
+  @Prop() disabled = false;
+
+  /** Whether or not the toggle button is currently on or off. Per default it is off. */
+  @Prop() state: boolean = false;
```

Comment:
If the prop has a default value, the type can be inferred by Stencil

Reply:
done